Frameworks/Foundation/NSKeyValueCoding.mm

```diff
@@ -578,30 +602,80 @@ - (NSMutableOrderedSet*)mutableOrderedSetValueForKeyPath:(NSString*)keyPath {
     return StubReturn();
 }
 
-/**
- @Status Stub
- @Notes
-*/
-- (void)setNilValueForKey:(NSString*)key {
-    UNIMPLEMENTED();
+static SEL KVCValidatorForPropertyName(NSObject* self, const char* key) {
+    size_t len = strlen(key);
+    // For the key "example", we must construct the following buffer:
+    // _ _ _ _ _ _ _ _ _ x a m p l e _ _ _ _ _ _ _ \0
+    // and fill it with the following characters:
+    // v a l i d a t e E x a m p l e : e r r o r : \0
+    char* chars = (char*)_alloca(8 + len + 8);
+    strcpy_s(chars + 9, len, key + 1);
+    chars[0] = 'v';
+    chars[1] = 'a';
+    chars[2] = 'l';
+    chars[3] = 'i';
+    chars[4] = 'd';
+    chars[5] = 'a';
+    chars[6] = 't';
+    chars[7] = 'e';
+    chars[8] = toupper(key[0]);
+    chars[8 + len] = ':';
+    chars[8 + len + 1] = 'e';
+    chars[8 + len + 2] = 'r';
+    chars[8 + len + 3] = 'r';
+    chars[8 + len + 4] = 'o';
+    chars[8 + len + 5] = 'r';
+    chars[8 + len + 6] = ':';
+    chars[8 + len + 7] = '\0';
+    SEL sel = sel_getUid(chars);
+    return ([self respondsToSelector:sel]) ? sel : nullptr;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
+ @Notes Invoked validate method must return BOOL
 */
 - (BOOL)validateValue:(id _Nullable*)ioValue forKey:(NSString*)key error:(NSError* _Nullable*)outError {
-    UNIMPLEMENTED();
-    return StubReturn();
+    SEL validationSelector = KVCValidatorForPropertyName(self, [key UTF8String]);
+    if (validationSelector) {
+        NSMethodSignature* sig = [self methodSignatureForSelector:validationSelector];
+        // 4 arguments: self, selector, ioValue, outError
+        if (sig.numberOfArguments == 4) {
+            // Reference platform does not check types, but will segfault when trying to return a value with size larger than BOOL
+            // And we will run into issues with values smaller than BOOL so simply ignore non-BOOL sized returns
+            if (sig.methodReturnLength == sizeof(BOOL)) {
+                return ((BOOL(*)(id, SEL, id*, NSError**))objc_msgSend)(self, validationSelector, ioValue, outError);
+            } else {
+                // But give a warning because this is stricter than the reference platform
+                TraceWarning(TAG,
```

Comment:
I'm not sure that would fit the contract of the method.  As for testing validation the NSObject_ValidateValueForKeyPath::Test validates the validation method being called